src/contracts/libraries/GPv2SafeERC20.sol

```diff
@@ -83,9 +85,37 @@ library GPv2SafeERC20 {
         // <https://docs.soliditylang.org/en/v0.7.6/internals/layout_in_memory.html>
         // solhint-disable-next-line no-inline-assembly
         assembly {
+            /// @dev Revert with an ABI encoded Solidity error with a message
+            /// that fits into 32-bytes.
+            ///
+            /// An ABI encoded Solidity error has the following memory layout:
+            ///
+            /// ------------+----------------------------------
+            ///  byte range | value
+            /// ------------+----------------------------------
+            ///  0x00..0x04 |        selector("Error(string)")
+            ///  0x04..0x24 |      string offset (always 0x20)
+            ///  0x24..0x44 |                    string length
+            ///  0x44..0x64 | string value, padded to 32-bytes
+            function revertWithMessage(length, message) {
```

Comment:
What happens if length > 32?

Reply:
> but this increases gas costs for tokens that don't return on transfer, right?

Correct.

Reply:
It won't inherently do anything too bad, just return garbage memory.